tests/src/JIT/opt/Devirtualization/simple.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+// Some simple interface call devirtualization cases
+
+interface Ix
+{
+    int F();
+}
+
+interface Iy
+{
+    int G();
+}
+
+public class B : Iy, Ix
+{
+    public int F() { return 44; }
+    virtual public int G() { return 55; }
+}
+
+public class Z : B
+{
+    new public int F() { return 33; }
+    override public int G() { return 22; }
+
+    static int Fx(Ix x) { return x.F(); }
+    static int Gy(Iy y) { return y.G(); }
+
+    public static int Main()
+    {
+        //       B:F            Z:G            B:F           B:G
+        return Fx(new Z()) + Gy(new Z()) + Fx(new B()) + Gy(new B()) - 65;
```

Comment:
I'll look into this. If C# can elide emitting an actual cast that version should also lead to devirtualization.

We currently lose types in checkcast. I have a partial fix worked up but need to understand (if downcasts are explicit) which casts add type info and which casts lose type info and avoid losing type info.